src/main/java/classify/user/InputParsing.java

```diff
@@ -115,6 +126,47 @@ public static void parseUserCommand(String[] userCommand, ArrayList<Student> mas
         }
     }
 
+    private static boolean isValidChoice(ArrayList<Student> masterStudentList, String input, boolean validChoice) {
+        try {
+            int choice = Integer.parseInt(input);
+            switch (choice) {
+            case 1:
+                listStudentsByName(masterStudentList);
+                validChoice = true;
+                break;
+
+            case 2:
+                listStudentsByTotalClasses(masterStudentList);
+                validChoice = true;
+                break;
+
+            default:
+                Ui.println(INVALID_CHOICE);
+            }
+        } catch (NumberFormatException e) {
```

Comment:
Solid use of exception to ensure that program does not crash if the user misinputs. LGTM!